### ISSUE TITLE
Sub-setting from a single DE file.

### ISSUE THREAD

**Issue by Peter-Simpson** [Bill-Gray/jpl_eph#12]
Hi Bill,

Thanks for creating this utility, it's saved me a lot of work! Fifteen year s ago, and with no Linux experience, I had to get FORTRAN running on a Windows PC when I needed to create a binary ephemeris file for the [ASCOM Platform ](https://github.com/ASCOMInitiative) that I support.

I'm trying to create a sub-set of the ascp1900.421 DE421 ephemeris file to cover just the years 2020 to 2050 rather than 1900 to 2050 and thought that the asc2eph -r2020,2050 parameter that filters output records would do the trick, but it only seems to work when the ASCII data is supplied as a collection of small files, some of which can be ignored if they are outside the required date range. 

I have managed to do what I want by changing asc2eph line 644 to read:

`for (year = 1900; year <= year_end; year += year_step)`

which processes the ascp1900.421 file and outputs the subset I'm looking for, but this is obviously a kludge.

I've tested the resulting binary file file using the NOVAS 3.1 planet ephemeris functions and the ephemeris data corresponds to the expected date range and the RA/Dec values agree to within a second or so of the USNO web site predictions. However, I'm concerned that there may be side effects that I'm not .aware of. 

Do you think this approach is OK and would you consider adding a feature to allow sub-setting regardless of whether the data is in a single file or multiple files?

Best wishes, Peter

**Comment by Bill-Gray** [comment]
Hi Peter,

Interesting.  I've always run with multiple files (and there is an `ascp2050.421` available).  Looking at the code,  though,  I think I can see why running with only one file.

"Normally",  DE421 would be provided as two files, `ascp1900.421` and `ascp2050.421`.  In that case,  the `determine_year_range()` function would (eventually) find both files and set `*year_start = 1900`, `*year_end = 2050`,  and `*year_step = 150`.  And similarly with any other multi-file case.

With just one file,  `*year_step` is left at the default value of five years,  and both `*year_start` and `*year_end` are left at 1900.  So the program figures it has data covering 1900 to 1905 and thinks : well,  I don't have anything that covers your desired time span.

I see two possible fixes : 

(1) Make sure both files are available,  even though only one is going to get used.  Requires no changes to the code.

(2) Add code to say that if only one file is found,  `*year_step = 1000;`  That way,  if only one file is found,  the program says : I don't really know what time span each ASCII file covers,  so let's set it to be really long.  Worst that'll happen is that we'll read in the entire file and realize that none of it covers the desired time span.

I've implemented the latter and pushed it to GitHub.  Shortly after doing so,  I saw that I'd made it more complicated than it needed to be;  all that was really required was
```c
    if( 1 == n_found)       /* year step wasn't actually determined */
       *year_step = 10000;
```
I'll probably fix that,  but it'll work Just Fine as it is.

BTW,  a note of thanks for your work on ASCOM.  I am now out of the desktop planetarium business (my Guide 9.1 software is getting increasingly obsolete).  But when I was in it,  the ability to ignore the bazillion different telescope control protocols saved me a lot of trouble.  I had already added code to control various Meade,  Celestron,  and other telescopes... just enough to make me perfectly happy to delegate further telescopes to ASCOM.
